pkg/cluster/cluster.go

```diff
@@ -138,6 +138,16 @@ func (c *Cluster) prepareBackupAndRestore() error {
 		}
 	}
 
+	// We use backup RS to mark that we have done backup creation and even restore process before.
```

Comment:
I still do not understand.

You would first check RS. If RS exists, you will not reach this func at all. If RS does not exist, you will enter this func. It has nothing to do with backup == nil.

But anyway, I think #435 is better to solve this issue.